corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowserEditor.java

```diff
@@ -516,52 +523,81 @@ public CorfuDynamicRecord editRecord(String namespace, String tableName,
+                " in table " + tableName + " and namespace " + namespace +
+                ".  Stream Id " + streamUUID);
 
+        CorfuTable<CorfuDynamicKey, CorfuDynamicRecord> table =
+                getTable(namespace, tableName);
         TableName tableNameProto = TableName.newBuilder().setTableName(tableName)
                 .setNamespace(namespace).build();
 
         Any defaultKeyAny =
                 dynamicProtobufSerializer.getCachedRegistryTable().get(tableNameProto)
                         .getPayload().getKey();
-        DynamicMessage keyMsg =
-                dynamicProtobufSerializer.createDynamicMessageFromJson(defaultKeyAny,
-                        keyToDelete);
-
-        CorfuDynamicKey dynamicKey =
-                new CorfuDynamicKey(defaultKeyAny.getTypeUrl(), keyMsg);
-        CorfuTable<CorfuDynamicKey, CorfuDynamicRecord> table =
-                getTable(namespace, tableName);
-        int numKeysDeleted = -1;
+        int numKeysDeleted = 0;
         try {
+            int recordsInTxn = 0;
             runtime.getObjectsView().TXBegin();
-            if (!table.containsKey(dynamicKey)) {
-                System.out.println("Key "+keyToDelete+" not found in "+fullName);
-                runtime.getObjectsView().TXEnd();
-                numKeysDeleted = 0;
-                return numKeysDeleted;
+            for (String keyToDelete: keysToDelete) {
+                DynamicMessage keyMsg =
+                        dynamicProtobufSerializer.createDynamicMessageFromJson(defaultKeyAny,
+                                keyToDelete);
+
+                CorfuDynamicKey dynamicKey =
+                        new CorfuDynamicKey(defaultKeyAny.getTypeUrl(), keyMsg);
+                if (!table.containsKey(dynamicKey)) {
+                    System.out.println("WARNING Key " + keyToDelete + " not found in " + fullName);
+                    continue;
+                }
+                numKeysDeleted++;
+                System.out.println(numKeysDeleted + ": Deleting record with Key " + keyToDelete);
+                table.delete(dynamicKey);
+                recordsInTxn++;
+                if (recordsInTxn > maxRecordsInBatch) {
+                    System.out.println("\nTransaction batch size "+maxRecordsInBatch+" hit, committing transaction\n");
+                    runtime.getObjectsView().TXEnd();
+                    recordsInTxn = 0;
+                    runtime.getObjectsView().TXBegin();
+                }
             }
-            System.out.println("Deleting record with Key " + keyToDelete +
-                    " in table " + tableName + " and namespace " + namespace +
-                    ".  Stream Id " + streamUUID);
-            table.delete(dynamicKey);
             runtime.getObjectsView().TXEnd();
-            System.out.println("\n======================\n");
-            numKeysDeleted = 1;
+            System.out.println("\n "+numKeysDeleted+" records deleted successfully.\n");
```

Comment:
We cannot assume that the records actually deleted will be the same as the size of the array passed in. Consider the case where a user runs the command and then kills the run half way through and then retries. Now half the records won't even show up in the table and the records deleted would be half the size of the array. So thats why the numKeysDeleted is computed within the loop and only incremented if the key is actually found in the table.

Reply:
Yes it cannot be the size of array passed. 
But it could be calculated after each TXEnd() like this -> numKeysDeleted += recordsInTxn. Because ultimately it's deleted only after it commits.

Reply:
done, thanks